olmo/optim.py

```diff
@@ -480,7 +481,11 @@ def get_max_grad_norm_ratio(
         return self._get_max_grad_norm_coeff(initial_max_grad_norm_ratio, step, max_steps)
 
     def _linear_warmup(self, initial_lr: float, step: int, warmup_steps: int = 2000) -> float:
-        return initial_lr * (0.1 + 0.9 * min(step, warmup_steps) / warmup_steps)
+        if self.warmup_min_lr is not None:
+            assert initial_lr > self.warmup_min_lr
+            return self.warmup_min_lr + (initial_lr - self.warmup_min_lr) * min(step, warmup_steps) / warmup_steps
+        else:
```

Comment:
From a cleaner code perspective, maybe it would be better to have just 1 formula and use `self.warmup_min_lr` to set the starting lr. Something like:
```
warmup_min_lr = self.warmup_min_lr if self.warmup_min_lr is not None else 0.1 * initial_lr
...
return warmup_min_lr + (initial_lr - warmup_min_lr) * min(step, warmup_steps) / warmup_steps
```

Reply:
Done: 75eba56a